assets/javascript/app.js

```diff
@@ -0,0 +1,93 @@
+// namespace object for food-finding functions
+var rest_test = {
+  
+  restaurants: [{
+    name: "McDonald's",
+    image:"url",
+    foodtype:"American",
+    description:"fast food",
+    link:"url"
+  }],
+
+  getNearbyRestaurants: function() {
+    return restaurants[Math.floor(Math.random()*restaurants.length)];
+  },
+
+  setFoodClick: function() {
+    $("#findfood").click(function() {
+      var resultsBody = $("<div>");
+  
+      var resultsName = $("<h2>").text(getNearbyRestaurants().name);
+      resultsName.addClass("");
+  
+      var resultsImage = $("<p>").text(getNearbyRestaurants().image);
+      resultsImage.addClass("");
+  
+      var resultsFood = $("<p>").text(getNearbyRestaurants().foodtype);
+      resultsFood.addClass("");
+  
+      var resultsDescription = $("<p>").text(getNearbyRestaurants().description);
+      resultsDescription.addClass("");
+  
+      var resultsLink = $("<p>").text(getNearbyRestaurants().link);
+      resultsLink.addClass("");
+  
+      resultsBody.append(resultsName, resultsImage, resultsFood, resultsDescription, resultsLink);
+      $('#results').append(resultsBody);
+    });
+  },
+
+  getNearbyLoc: function() {
+    $.ajax({
+      method: "GET",
+      crossDomain: true,
+      // count is how many restaurants we want
+      // establishment_type is the type of restaurant we want
+      url: "https://developers.zomato.com/api/v2.1/search?count=10&lat=" + loc.lat + "lon=" + loc.lon + "&radius=2",
+      dataType: "json",
+      async: true,
+      headers: {
+        "user-key": "7e43dd3df445b7aee59f4e05cf1204c7"
+      },
+  
+      success: function(data) {
+          console.log(data);
+        var res = [];
+        res = data.restaurants;
+        for (var j = 0; j < res.length; j++) {
+      
+            $('.results').append("<div class='name'>" + "Restaurant: " + res[j].restaurant.name + "</div>" + "\n" + "<div class='cuisines'>" + res[j].restaurant.cuisines + "</div>");
+    
+        }
+      },
+      
+      error: function() {
+        $('.results').append("<div>Sorry, data is not coming through. Refresh and try again.</div>");
+      }
+    });
+  }
+}
+
+// namespace object for location functions
+var loc = {
```

Comment:
A couple things:
- loc is just a namespace object. Since it's unlikely that we'll have two functions called getLocation in this project, we don't really need it.
- I kept lat and lon outside the function because watchPosition is a listener that modifies vars lat and lon every time the user moves or the browser gets a more precise lock (gps can take a minute to activate). Your suggestion would only return a location object when getLocation is called, which may be OK.